crates/runtime/tests/rehydration/mod.rs

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+//! Runs federation integration tests for `MySQL`.
+//!
+//! Expects a Docker daemon to be running.
+use crate::{
+    docker::RunningContainer,
+    get_test_datafusion,
+    mysql::common::{get_mysql_conn, make_mysql_dataset, start_mysql_docker_container},
+    runtime_ready_check,
+};
+use std::sync::Arc;
+
+use crate::init_tracing;
+
+use app::AppBuilder;
+use arrow::array::RecordBatch;
+use datafusion_table_providers::sql::arrow_sql_gen::statement::{
+    CreateTableBuilder, InsertBuilder,
+};
+use futures::TryStreamExt;
+use mysql_async::{prelude::Queryable, Params, Row};
+use runtime::{datafusion::query::Protocol, spice_data_base_path, status, Runtime};
+use spicepod::component::dataset::{
+    acceleration::{Acceleration, Mode},
+    Dataset,
+};
+use tracing::instrument;
+
+/// Validates spill-to-disk and rehydration functionality by simulating runtime restarts
+/// and checking data consistency.
+///
+/// 1. Retrieve num of rows using a native MySQL connection to use as a baseline
+/// 2. Start Spice, retrieve row count and top 100 items after acceleration is completed, compare with baseline
```

Comment:
Is the top 100 items retrieval going to be included in this PR

Reply:
Good catch - updated 👍